src/main/java/org/jenkinsci/maven/plugins/hpi/RunMojo.java

```diff
@@ -419,12 +408,16 @@ private void generateHpl() throws MojoExecutionException, MojoFailureException {
     public void configureWebApplication() throws Exception {
         // Jetty tries to do this in WebAppContext.resolveWebApp but it failed to delete the directory.
         File extractedWebAppDir= new File(getTmpDirectory(), "webapp");
-        if (isExtractedWebAppDirStale(extractedWebAppDir, webApp)) {
+        if (isExtractedWebAppDirStale(extractedWebAppDir, webAppFile)) {
             FileUtils.deleteDirectory(extractedWebAppDir);
         }
         
         super.configureWebApplication();
-        getWebApplication().setWebAppSrcDir(webApp);
+        getWebAppConfig().setWar(webAppFile.getCanonicalPath());
+        // cf. https://wiki.jenkins-ci.org/display/JENKINS/Jetty
+        HashLoginService hashLoginService = (new HashLoginService("Jenkins Realm"));
+        hashLoginService.setConfig(System.getProperty("jetty.home", ".") + "/etc/realm.properties");
```

Comment:
I doubt this would be used in practice anyway, but indeed this would be better set to `work`.